### PR TITLE
Add TypeScript declarations to iron-location.

### PR DESCRIPTION
**Pull request by aomarks**
This PR adds TypeScript declarations generated by https://github.com/Polymer/gen-typescript-declarations/

These declarations can be re-generated by running `npm run update-types`.

Tracker: https://github.com/Polymer/gen-typescript-declarations/issues/79

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,16 @@
+{
+  "name": "@polymer/iron-location",
+  "private": true,
```
Comment (abdonrd) [inline package.json]: Why `private: true`? I understand that the idea is to be public in npm.

File: package.json [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  "description": "Bidirectional data binding into the page's URL.",
+  "repository": {
```
Comment (abdonrd) [inline package.json]: Short proposal:

```
"repository": "PolymerElements/iron-location",
```

File: package.json (hunk 1 above)
Comment (rictic) [reply package.json]: This is separate from the 3.0 move to npm. It'll be public when this code is on npm, this is just adding typings.

File: package.json (hunk 2 above)
Comment (aomarks) [reply package.json]: This is just copied from bower. Doesn't seem to be any advantage to changing it.

https://docs.npmjs.com/files/package.json#repository

**Pull request merged**